tests/improc/test_zogy.py

```diff
@@ -0,0 +1,436 @@
+    sim = Simulator(
+        image_size_x=imsize,  # not too big, but allow some space for stars
+        background_std=0,  # keep background constant between images
+        background_minimum=0.01,  # allow very low background for the reference image
+        read_noise=0,  # make it easier to input the background noise (w/o source noise)
+        seeing_mean=2.0,  # make the seeing a little more pronounced
+        seeing_std=0,  # keep seeing constant between images
+        seeing_minimum=0.1,  # assume seeing never gets much better than this
+        transmission_std=0,  # keep transmission constant between images
+        pixel_qe_std=0,  # keep pixel QE constant between images
+        gain_std=0,  # keep pixel gain constant between images
+        saturation_limit=1e9,  # make it impossible to saturate
+        star_number=num_stars,
+    )
+
+    sim.pars.seeing_mean = 0.7  # good seeing reference image
+    sim.pars.background_mean = 2.0  # low background reference image
+    sim.make_image(new_sky=True)
+    truth1 = sim.truth
+    im1 = sim.apply_bias_correction(sim.image)
+    im1 -= truth1.background_instance
+    psf1 = truth1.psf_downsampled
+    bkg1 = truth1.total_bkg_var
+
+    # add a few sources
+    fluxes = [100, 500, 1000, 2000, 5000, 10000, 50000, 100000]
+    pos = np.linspace(0, imsize, len(fluxes) + 1, endpoint=False)[1:]
+    for f, p in zip(fluxes, pos):
+        sim.add_extra_stars(flux=f, x=p, y=p)
+
+    sim.pars.seeing_mean = 2.5  # seeing is much worse in new image
+    sim.pars.background_mean = 30.0  # background is much higher in new image
+
+    iterations = 10
+    expected = np.zeros((iterations, len(fluxes)))
+    measured = np.zeros((iterations, len(fluxes)))
+
+    for j in range(iterations):
+        sim.make_image(new_sky=True)
+        truth2 = sim.truth
+        im2 = sim.apply_bias_correction(sim.image)
+        im2 -= truth2.background_instance
+        psf2 = truth2.psf_downsampled
+        bkg2 = truth2.total_bkg_var
+
+        # need to figure out better values for this
+        F1 = 1.0
+        F2 = 1.0
+
+        zogy_diff, zogy_psf, zogy_score, zogy_score_corr, alpha, alpha_err = zogy_subtract(
+            im1, im2, psf1, psf2, np.sqrt(bkg1), np.sqrt(bkg2), F1, F2,
+        )
+        edge = 8  # do not trigger on stars too close to the edge
+        S = np.ones(zogy_score_corr.shape) * np.nan
+        S[edge:-edge, edge:-edge] = zogy_score_corr[edge:-edge, edge:-edge].copy()
+        B = truth2.background_instance
+        P = zogy_psf
+
+        for i, f in enumerate(fluxes):
+            x = y = int(pos[i])
+            c = S[y-edge:y+edge, x-edge:x+edge]
+            expected[j, i] = f * np.sqrt(np.sum(P ** 2) / B)
+            expected[j, i] *= np.sqrt(B / (B + f * np.sum(P ** 2)))  # add the Poisson noise from the source
+            measured[j, i] = np.nanmax(c)
+
+    mean = np.nanmean(measured, axis=0)
+    err = np.nanstd(measured, axis=0)
+    exp = expected[0, :]
```

Comment:
Why are you comparing the means of the measured values to the first expected value?  If the background_instance is different in each expected value, shouldn't you compare each measured to each expected, or the mean of measured to the mean of expected?

Reply:
The expected is calculated and is the same for each `j` (each iteration), and then we average the measured (which is different for each iteration) but choose only the first of each expected. I tried to remove the superfluous calculation but it made the code more complicated because of the order of the loops I am using.

Reply:
In this case the background_instance is the same all the time, but if we ever change that, we will need to use an average. I will put the expected values as a mean to make it more robust.